frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/rbac.tsx

```diff
@@ -0,0 +1,147 @@
+  return newImpersonate ? `${newImpersonate.kind}~{impersonate.user}` : '';
+};
+
+/**
+ * Provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Object with resource access information.
+ * * */
+export const checkAccess = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?,
+): Promise<SelfSubjectAccessReviewKind> => {
+  // Destructure the attributes with defaults so we can create a stable cache key.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes || {};
+  return checkAccessInternal(
+    group,
+    resource,
+    subresource,
+    verb,
+    name,
+    namespace,
+    getImpersonateKey(impersonate),
+  );
+};
+
+/**
+ * Hook that provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Array with isAllowed and loading values.
+ * * */
+export const useAccessReview2 = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?,
+): [boolean, boolean] => {
+  const [loading, setLoading] = useSafetyFirst(true);
+  const [isAllowed, setAllowed] = useSafetyFirst(false);
+  // Destructure the attributes to pass them as dependencies to `useEffect`,
+  // which doesn't do deep comparison of object dependencies.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes;
+  const impersonateKey = getImpersonateKey(impersonate);
+  React.useEffect(() => {
+    checkAccessInternal(group, resource, subresource, verb, name, namespace, impersonateKey)
+      .then((result: SelfSubjectAccessReviewKind) => {
+        setAllowed(result.status.allowed);
+        setLoading(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('SelfSubjectAccessReview failed', e);
+        // Default to enabling the action if the access review fails so that we
+        // don't incorrectly block users from actions they can perform. The server
+        // still enforces access control.
+        setAllowed(true);
+        setLoading(false);
+      });
+  }, [setLoading, setAllowed, group, resource, subresource, verb, name, namespace, impersonateKey]);
+
+  return [isAllowed, loading];
+};
+
+/**
+ * Hook that provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return The isAllowed boolean value.
+ * * */
+export const useAccessReview = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?,
+): boolean => useAccessReview2(resourceAttributes, impersonate)[0];
```

Comment:
when moving to SDK do we need to have both `useAccessReview` and `useAccessReview2` in SDK or can just have superset i.e `useAccessReview2` and rename it to `useAccessReview`?

@florkbr / @andrewballantyne do you have any opinion on it? and should we expose it via APIs

Reply:
Added my feedback here: https://github.com/openshift/console/pull/10847#discussion_r799626195